File: src/main/java/org/noxet/noxetserver/util/TextBeautifier.java
package org.noxet.noxetserver.util;

public class TextBeautifier {
    private static final String beautifulAlphabetCharacters = "ᴀʙᴄᴅᴇꜰɢʜɪᴊᴋʟᴍɴᴏᴘꞯʀꜱᴛᴜᴠᴡxʏᴢ";

    public static String beautify(String uglyText) {
        return beautify(uglyText, true);
    }

    public static String beautify(String uglyText, boolean formatUppercase) {
        StringBuilder beautifulText = new StringBuilder();

        for(char character : uglyText.toCharArray()) {
            if(character >= 'a' && character <= 'z')
                beautifulText.append(beautifulAlphabetCharacters.charAt(character - 'a'));
            else if(formatUppercase && character >= 'A' && character <= 'Z')
                beautifulText.append(beautifulAlphabetCharacters.charAt(character - 'A'));
            else
                beautifulText.append(character);
        }

        return beautifulText.toString();
    }
}

File: src/main/java/org/noxet/noxetserver/util/TextBeautifier.scala
package org.noxet.noxetserver.util

object TextBeautifier:
    private val beautifulAlphabetCharacters = "ᴀʙᴄᴅᴇꜰɢʜɪᴊᴋʟᴍɴᴏᴘꞯʀꜱᴛᴜᴠᴡxʏᴢ"

    def beautify(uglyText: String): String = beautify(uglyText, true)

    def beautify(uglyText: String, formatUppercase: Boolean = true): String =
        val beautifulText = StringBuilder()

        for
          character <- uglyText.toCharArray
        do
          beautifulText.append(
            if character >= 'a' && character <= 'z' then
              beautifulAlphabetCharacters.charAt(character - 'a')
            else if formatUppercase && character >= 'A' && character <= 'Z' then
              beautifulAlphabetCharacters.charAt(character - 'A')
            else character
          )

        beautifulText.toString